File: html/ttt/js/tic-tac-toe.js
let current_player, symbol, game_over, winner, tie, move_count, game_started

function reset_gameboard() {
    let columns = document.getElementById('gameboard').children;
    for (let squares of columns) {
        for (let square of squares.children) {
            square.innerHTML = "";
        }
    }
}

function initiate_game() {
    current_player = "";
    symbol = "";
    game_over = false;
    winner = "";
    tie = false;
    move_count = 0;
    game_started = false;

    reset_gameboard();

    Math.random() < 0.5 ? symbol = "X" : symbol = "O";
    if (symbol === "X") {
        current_player = "x-player";
    } else {
        current_player = "o-player";
    }

    let players = document.getElementById('players').children;
    for (let player of players) {
        if (player.id === current_player) {
            player.setAttribute('class', 'border-b-2 border-black active');
        } else {
            player.setAttribute('class', '')
        }
    }
}

function switch_player() {
    document.getElementById(current_player).setAttribute('class', '');

    if (current_player === "x-player") {
        current_player = "o-player";
        symbol = "O";
    } else {
        current_player = "x-player";
        symbol = "X";
    }

    document.getElementById(current_player).setAttribute('class', 'border-b-2 border-black active');
}

function place_symbol(square) {
    document.getElementById(square).innerHTML = symbol;
}

function start_game() {
    initiate_game();
    game_started = true;
    document.getElementById('game-status').innerHTML = "Game Started";
    document.getElementById('reset-game').removeAttribute('disabled');
    document.getElementById('reset-game').addEventListener('click', reset_game);
}

function reset_game() {
    initiate_game();
    for (let player of document.getElementById('players').children) {
        player.setAttribute('class', '');
    }
    document.getElementById('game-status').innerHTML = "Press Start Game Again";
}

function update_gameboard(e) {
    if (!game_started || game_over) {
        return;
    }

    if (document.getElementById(e.target.id).innerHTML === '') {
        place_symbol(e.target.id);
        switch_player();
        move_count++;
        if (move_count === 9) {
            game_over = true;
            tie = true;
            return;
        }

        check_for_winner();
    }
}

function check_for_winner() {
    function check_rows() {

    }
    function check_columns() {

    }

    function check_diagonals() {

    }
}

document.getElementById('gameboard').addEventListener('click', update_gameboard);

document.getElementById('start-game').addEventListener('click', start_game);

File: python/ttt/tictactoe.py
import tkinter as tk
import random

# Create the window named "window" and disable resizing
window = tk.Tk()
window.minsize(800, 900)
window.resizable(False, False)

# Create the outer frame that will hold all other widgets
game = tk.Frame(master=window, background="#66B2FF")
game.pack(fill=tk.BOTH, expand=True)

# Global variables
current_player = ""
symbol = ""
game_over = False
winner = ""
tie = False
move_count = 0
game_started = False
game_status = tk.StringVar()


def initialize_game():
    current_player = ""
    symbol = ""
    game_over = False
    winner = ""
    tie = False
    move_count = 0
    game_started = False

    reset_gameboard()

    random.Random(random.seed(float)).random()


def reset_gameboard():
    pass


def switch_player():
    pass


def place_symbol():
    print("Symbol placed")


def start_game(event):
    game_status.set("Game started")


def reset_game(event):
    game_status.set("Game reset")


def update_gameboard():
    pass


def check_for_winner():
    def check_rows():
        pass

    def check_columns():
        pass

    def check_diagonals():
        pass

    check_rows()
    check_columns()
    check_diagonals()


def build_game():
    # frame for players
    players_frame = tk.Frame(master=game)
    players_frame.grid(row=0, column=1)
    oplayer = tk.Frame(master=players_frame, borderwidth=1)
    oplayer_label = tk.Label(master=oplayer, text="O", font=("Monospace", 40))
    oplayer_label.pack()
    oplayer.grid(row=0, column=0)

    xplayer = tk.Frame(master=players_frame, borderwidth=1)
    xplayer_label = tk.Label(master=xplayer, text="X", font=("Monospace", 40))
    xplayer_label.pack()
    xplayer.grid(row=0, column=2)

    # frame for gameboard
    gameboard_frame = tk.Frame(master=game, borderwidth=1)
    gameboard_frame.grid(row=1, column=0)
    for i in range(3):
        for j in range(3):
            gameboard = tk.Frame(master=gameboard_frame, borderwidth=1)
            gameboard.grid(row=i, column=j)
            gameboard_label = tk.Label(master=gameboard, text="X", font=("Monospace", 40))
            gameboard_label.pack()

    game_meta_frame = tk.Frame(master=game, borderwidth=1)
    game_meta_frame.grid(row=2, column=0)
    # frame for status and controls
    game_status_text = tk.Label(master=game_meta_frame, text="...")
    game_status_text.grid(row=1, column=1)
    game_status_text["textvariable"] = game_status

    start_button = tk.Button(master=game_meta_frame, text="Start Game")
    start_button.bind("<Button-1>", start_game)
    start_button.grid(row=2, column=0)

    reset_button = tk.Button(master=game_meta_frame, text="Reset Game")
    reset_button.bind("<Button-1>", reset_game)
    reset_button.grid(row=2, column=2)


build_game()

# Start the window's event loop
window.mainloop()
